File: examples/go/.hm/pipeline.ts
import { pipeline, push, type PipelineDefinition } from "@harmont/hm";
import { go } from "@harmont/hm/toolchains";

const project = go({ path: "." });

const pipelines: PipelineDefinition[] = [
  {
    slug: "ci",
    triggers: [push({ branch: "main" })],
    pipeline: pipeline(
      [project.build(), project.test(), project.vet(), project.fmt()],
      { env: { CI: "true" }, defaultImage: "ubuntu:24.04" },
    ),
  },
];

export default pipelines;

File: examples/elixir/.hm/pipeline.py
"""Elixir example pipeline."""
from __future__ import annotations

import harmont as hm


@hm.pipeline(
    "ci",
    env={"CI": "true", "MIX_ENV": "test"},
    triggers=[hm.push(branch="main")],
)
def ci() -> tuple[hm.Step, ...]:
    project = hm.elixir(path=".")
    return (
        project.compile(),
        project.test(),
        project.format(),
        project.credo(),
        project.dialyzer(),
        project.deps_audit(),
        project.hex_audit(),
    )
